OneCommunityGlobal/HighestGoodNetworkApp
# Abdel deadline follow up tracking solution

# Description
1.	#### $\textcolor{red}{\textsf{(PRIORITY HIGH) Jae:}}$  Create way for me to track people I’ve followed up with about deadlines 

-  On the Tasks tab, I’d like a little box I can click on and off. I’m thinking the best location is to the right of the time numbers below. This is because it is related to those numbers and I want it super easy to scroll the whole tasks list and see who needs a followup or not.
- It should be visible and usable by every class but Volunteers. Any class checking the box will change its appearance for all others that can see it. This is so everyone knows if a person has been followed up with yet.
- It should have mouseover text that says, “This box is used to track follow ups. Clicking it means you’ve checked in with a person that they are on track to meet their deadline.”
- I’d like the box to be Red outline with light pink filler when a person is at 50% of their task deadline, again at 75% of their task deadline, and again when they are at 90% of their task deadline. This is to make it clear that it needs to be checked/the person needs to be followed up with. Once checked, the box should change back to normal but with a check in it. 
- I’d like the box to reset/auto-clear the check within the box when a person reaches 75% and 90%, so the box is ready/needs to be checked again. This is so I can track when I’ve followed up with people and also know when/if my managers are following up with people.
- I’d like an “i” icon for information that explains all of the information above.
- I’d also like mouseover text that will be different for the box at each level 
      **A -**. 0-50%: Check this box once you’ve checked in for the first time with this team member to make sure they are clear on their task.
      **B -**. 0-50%: Check this box once you’ve checked in for the first time with this team member to make sure they are clear on their task.
      **C -**. Your team member’s task should be at least 75% complete! Check this box once you’ve confirmed they are on track to meet their deadline. Request additional time be added to their task if it is needed.
      **D -**. Your team member’s task should be almost complete! Check this box once you’ve confirmed they are on track to meet their deadline. Request additional time be added to their task if it is needed.
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49083865/1862b0c7-1064-4291-bfee-dc1cb8f05b48)

## Related PRS:

- This frontend PR is related to the [419](https://github.com/OneCommunityGlobal/HGNRest/pull/419) backend PR.

## Main changes explained:

- Update TeamMemberTask file Add handleCheckboxFollowUp and followUpMouseoverText
- Update TeamMemberTasks file Add handleFollowUp.
- Update TeamMemberTasks/task file Add setFollowUp and checkWhoNeedsFollowUp.
- Update TeamMemberTasks/reducer file Add SET_FOLLOWED_UP.

## How to test:

- check into the current branch.
- do `npm run start:local` to run this PR locally.
- Create a task and assign it to a user. 
- Test for the requirements in the description.

To create a Task for a user, you must be logged in as an Admin or Owner and the individual to be given the task must be assigned the Project the task will be/is a part of. Then go to Other Links—>Projects—>WBS—>Choose the WBS—> Assign the task or “Add Task”  

Here’s a video tutorial too: https://www.loom.com/share/8bc672c323b64955aac4bce46c3794aa?sid=b05f5604-87cb-4ca1-80b8-49bd823893ee  

## Test:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49083865/9866e33f-6de4-4c81-99e2-779ee6fd178b


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49083865/0971f0cb-3a03-492e-9621-f6f0b08184fb



## Note:
To observe the changes on the checkbox after adding time to a user, do a hard refresh:
Windows/Linux: Ctrl + F5
Mac: Command + Shift + R
